plugins/es-repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3BlobContainer.java

```diff
@@ -259,7 +259,14 @@ void executeSingleUpload(final S3BlobStore blobStore,
         try (AmazonS3Reference clientReference = blobStore.clientReference()) {
             clientReference.client().putObject(putRequest);
         } catch (final AmazonClientException e) {
-            throw new IOException("Unable to upload object [" + blobName + "] using a single upload", e);
+            if (!this.blobStore.getRepositoryMetaData().settings().hasValue(S3RepositorySettings.ACCESS_KEY_SETTING.getKey()) &&
```

Comment:
Is the `executeSingleUpload` the right place for it? There are also delete, read, etc. methods.
Is it possible to do it somewhere in `S3ClientSettings`?